api/node/__test__/api.spec.ts

```diff
@@ -55,3 +61,49 @@ test('constructor parameters', (t) => {
     t.is(test.check, "test");
     t.is(hello, "hello");
 })
+
+
+test('callback closure cyclic references do not prevent GC', async (t) => {
+
+    // Setup:
+    // A component instance with a callback installed from JS:
+    //    * The callback captures the surrounding environment, which
+    //      includes an extra reference to the component instance itself
+    //      --> a cyclic reference
+    //    * Invoking the callback clears the reference in the outer captured
+    //      environment.
+    //
+    // Note: WeakRef's deref is used to observe the GC. This means that we must
+    // separate the test into different jobs with await, to permit for collection.
+    // (See https://tc39.es/ecma262/multipage/managing-memory.html#sec-weak-ref.prototype.deref)
+
+    let demo = loadFile(path.join(__dirname, "resources/test-constructor.slint")) as any;
+    let callback_invoked = false;
+    let demo_weak = new WeakRef(demo);
+
+    demo.my_callback = () => {
+        demo = null;
+        callback_invoked = true;
+    };
+
```

Comment:
What if you add something like this:

```suggestion

function scope() {
   let copy = demo;
   copy.say_hello = () => {
        console.log(copy.check);
   };
}
scope();
```

i.e: add another callback that use a property from some other variable with a small scope.

Reply:
That did the trick, good idea. Pushed, now the test fails.